src/components/Sidebar/Sidebar.tsx

```diff
@@ -10,17 +11,22 @@ const dataKeys = Object.keys(data) ?? [];
 
 const Sidebar = () => {
 
+  const router = useRouter();
   const dispatch = useAppDispatch()
   const selectedItem = useAppSelector((state) => state.code.selectedItem)
 
   useEffect(() => {
     const firstKey = dataKeys.length > 0 ? dataKeys[0] : ''
-    const objectData = { ...(data as any)[firstKey][0], globalKey: firstKey, index: 0 }
+    const objectData = { ...data[firstKey][0], globalKey: firstKey, index: 0 }
```

Comment:
will this break?
if the data is `{}`, firstKey will results in ''
won't `{}[''][0]` break?

Reply:
This one is not solved yet I think
```
const firstKey = dataKeys.length > 0 ? dataKeys[0] : ''
// should add if here
if (!firstKey) return
// this way, the firstKey won't be '', and data[''][0] won't happen.
const objectData = { ...data[firstKey][0], globalKey: firstKey, index: 0 }
```